rest/handler/tracinghandler.go

```diff
@@ -48,7 +91,21 @@ func TracingHandler(serviceName, path string) func(http.Handler) http.Handler {
 
 			// convenient for tracking error messages
 			propagator.Inject(spanCtx, propagation.HeaderCarrier(w.Header()))
-			next.ServeHTTP(w, r.WithContext(spanCtx))
+			trw := &traceResponseWriter{
+				w:    w,
+				code: http.StatusOK,
+			}
+			next.ServeHTTP(trw, r.WithContext(spanCtx))
+
+			span.SetAttributes(attribute.KeyValue{
+				Key:   traceKeyStatusCode,
+				Value: attribute.IntValue(trw.code),
+			})
+			if trw.code >= http.StatusBadRequest {
+				span.SetStatus(codes.Error, http.StatusText(trw.code))
+			} else {
+				span.SetStatus(codes.Ok, "")
```

Comment:
Is `span.SetStatus(codes.Ok, http.StatusText(trw.code))` better?

Reply:
Yes, fixed.